### PR TITLE
Add github citation file

### PR DESCRIPTION
**Pull request by emiliom**
See https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files

I used this citation file as a guide: https://github.com/microsoft/torchgeo/blob/main/CITATION.cff

See my comment in the file about whether to refer to a specific echopype release.

cc @lsetiawan , since he reminded us of this capability :smiley:

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+message: "If you use this software, please cite it as below."
+authors:
+- family-names: "Lee"
+  given-names: "Wu-Jung"
+  orcid: "https://orcid.org/0000-0002-4112-2034"
+- family-names: "Mayorga"
+  given-names: "Emilio"
+  orcid: "https://orcid.org/0000-0003-2574-4623"
+- family-names: "Setiawan"
+  given-names: "Landung"
```
Comment (lsetiawan) [inline CITATION.cff:11]:
```suggestion
  given-names: "Landung"
  orcid: "https://orcid.org/0000-0002-1624-2667"
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  type: article
+  authors:
+  - family-names: "Lee"
+    given-names: "Wu-Jung"
+    orcid: "https://orcid.org/0000-0002-4112-2034"
+  - family-names: "Mayorga"
+    given-names: "Emilio"
+    orcid: "https://orcid.org/0000-0003-2574-4623"
+  - family-names: "Setiawan"
+    given-names: "Landung"
```
Comment (lsetiawan) [inline CITATION.cff:36]:
```suggestion
    given-names: "Landung"
    orcid: "https://orcid.org/0000-0002-1624-2667"
```

**Review by lsetiawan — APPROVED** [review]
Thanks Emilio! Please add my orcid, otherwise looks good to me 😄

**Comment by leewujung** [comment]
On the ORCID part: I included everyone's ORCID in the [v0.5.4 release](https://zenodo.org/record/5544192#.YZ1P7L3MIc9). @emiliom could you please include all of them? Thanks!

**Comment by emiliom** [comment]
I've added ORCIDs for everyone. It didn't occur to me that everyone would have ORCIDs!

**Pull request merged**